### PR TITLE
chore: deps: update ffi and go-state-types for NV23

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Update ffi and go-state-types for NV23

### CODE REVIEW

File: extern/filecoin-ffi
Comment (aarshkshah1992) [inline extern/filecoin-ffi:1]: @rvagg I believe this is all that is needed to update ffi.

Comment (rvagg) [reply extern/filecoin-ffi:1]: yep, looks good, pointing to e467d2992e3f9bd0 is what we want

**Comment by aarshkshah1992** [comment]
CI failure is a known flaky and has been documented at https://github.com/filecoin-project/lotus/issues/12001.

File: go.mod [hunk 1]
```diff
@@ -33,7 +33,7 @@ require (
 	github.com/elastic/gosigar v0.14.2
 	github.com/etclabscore/go-openrpc-reflect v0.0.36
 	github.com/fatih/color v1.15.0
-	github.com/filecoin-project/filecoin-ffi v0.30.4-0.20220519234331-bfd1f5f9fe38
```
Comment (rvagg) [inline go.mod:36]: I was scratching my head about this being so old until I remembered about the `replace`, so this isn't technically needed but certainly nice to get it off 2022! can't wait for ffi go binary modules tho.

**Review by rvagg — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (aarshkshah1992) [reply go.mod:36]: @rvagg I don't think I will be working on ffi go modules anytime soon man. It is a non trivial amount of work and I would like to focus on the ETH/Fevm side of things.

Comment (aarshkshah1992) [reply go.mod:36]: If someone else wants to hack on it -> happy to share my findings/direct them to the docs we've produced for it.

**Pull request merged**